### PR TITLE
module for creating, managing Infoblox NIOS nameserver groups

### PR DESCRIPTION
**Pull request by ebirn**
##### SUMMARY
Module to create and manage Infoblox nameserver groups through the Wapi nsgroups objects.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
nios_nsgroup

##### ANSIBLE VERSION

```
ansible 2.5.0 (nios_nsgroup 7cc47392f4) last updated 2018/02/08 18:37:52 (GMT +200)
  config file = None
  configured module search path = [u'/Users/erich.birngruber/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/erich.birngruber/projects/ansible_fork/lib/ansible
  executable location = /Users/erich.birngruber/projects/ansible_fork/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

### CODE REVIEW

File: lib/ansible/modules/net_tools/nios/nios_nsgroup.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nios_nsgroup
+short_description: manage InfoBlox DNS nameserver groups
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/nios/nios_nsgroup.py]: `Manage...`

File: lib/ansible/modules/net_tools/nios/nios_nsgroup.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/nios/nios_nsgroup.py:14]: Please ensure all the descriptions are full sentences (capital letters & full stops)

**Comment by ebirn** [comment]
@ansibot ready_for_review

**Comment by ebirn** [comment]
@ansibot bot_status

File: lib/ansible/modules/net_tools/nios/nios_nsgroup.py [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        name=dict(type='str', required=True, ib_req=True),
+        enable_preferred_primaries=dict(type='bool', default=False),
+        grid_replicate=dict(type='bool', default=False),
+        lead=dict(type='bool', default=False),
+        preferred_primaries=dict(type='list', elements='dict', options=extserver_spec, default=[]),
+        stealth=dict(type='bool', default=False),
+    )
+
+    ib_spec = dict(
+        name=dict(type='str', required=True, ib_req=True),
```
Comment (justjais) [inline lib/ansible/modules/net_tools/nios/nios_nsgroup.py:312]: The default is of `type=str`, hence not required to mention explicitly.

File: lib/ansible/modules/net_tools/nios/nios_nsgroup.py [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+    # cleanup tsig fields
+    def clean_tsig(ext):
+        if 'tsig_key' in ext and not ext['tsig_key']:
+            del ext['tsig_key']
+        if 'tsig_key' not in ext and 'tsig_key_name' in ext and not ext['tsig_key_name']:
+            del ext['tsig_key_name']
+        if 'tsig_key' not in ext and 'tsig_key_alg' in ext:
+            del ext['tsig_key_alg']
+
```
Comment (justjais) [inline lib/ansible/modules/net_tools/nios/nios_nsgroup.py:258]: Can you please give one-liner comment for each fn, like you have added in `clean-tsig`

File: lib/ansible/modules/net_tools/nios/nios_nsgroup.py [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+                          transform=grid_primary_preferred_transform),
+        grid_secondaries=dict(type='list', elements='dict', options=memberserver_spec,
+                              transform=grid_secondaries_preferred_primaries_transform),
+        external_primaries=dict(type='list', elements='dict', options=extserver_spec, transform=ext_primaries_transform),
+        external_secondaries=dict(type='list', elements='dict', options=extserver_spec,
+                                  transform=ext_secondaries_transform),
+        is_grid_default=dict(type='bool', default=False),
+        use_external_primary=dict(type='bool', default=False),
+        extattrs=dict(),
+        comment=dict(type='str'),
```
Comment (justjais) [inline lib/ansible/modules/net_tools/nios/nios_nsgroup.py:323]: The default is of `type=str`, so you can remove the explicit type-call.

**Comment by justjais** [comment]
@ebirn Thank you for your contribution, can you please write Integration and target tests for this module file.

**Comment by justjais** [comment]
@ebirn will you able to write the integration tests for the respective module and merge the PR, else I would like to close the PR and open new PR with respective and required changes so that it can be merged in `2.8`

**Comment by justjais** [comment]
Closing this PR and opening a new PR #49124 for making this module part of Ansible `2.8` release.

**Pull request closed without merging**